### PR TITLE
PUBDEV-8713: Export with_h2o_no_progress

### PR DESCRIPTION
**Pull request by tomasfryda**
https://h2oai.atlassian.net/browse/PUBDEV-8713

### CODE REVIEW

**Comment by ledell** [comment]
Rather than adding a new function, we can optionally allow users to pass in an expression to the `h2o.no_progress()` function.  Maybe let's add an optional arg called `expr` or similar?

File: h2o-r/h2o-package/R/communication.R [hunk 1]
```diff
@@ -1111,11 +1115,28 @@ h2o.is_client <- function() get("IS_CLIENT", .pkg.env)
...
+h2o.no_progress <- function(expr) {
+  if (missing(expr)) {
+    assign("PROGRESS_BAR", FALSE, .pkg.env)
+    invisible(NULL)
+  } else {
+    show_progress <- .h2o.is_progress()
+    if (length(show_progress) == 0L || show_progress) {
+      on.exit(h2o.show_progress())
+    } else {
+      on.exit(h2o.no_progress())
```
Comment (sebhrusen) [inline h2o-r/h2o-package/R/communication.R:1127]: unnecessary?

File: h2o-r/h2o-package/R/communication.R [hunk 2]
```diff
@@ -1138,7 +1159,21 @@ h2o.no_progress <- function() assign("PROGRESS_BAR", FALSE, .pkg.env)
...
 #' @export
-h2o.show_progress <- function() assign("PROGRESS_BAR", TRUE, .pkg.env)
+h2o.show_progress <- function(expr) {
+  if (missing(expr)) {
+    assign("PROGRESS_BAR", TRUE, .pkg.env)
+    invisible(NULL)
+  } else {
+    show_progress <- .h2o.is_progress()
+    if (length(show_progress) == 0L || show_progress) {
+      on.exit(h2o.show_progress())
```
Comment (sebhrusen) [inline h2o-r/h2o-package/R/communication.R:1169]: unnecessary?

**Review by sebhrusen — APPROVED** [review]

File: h2o-r/h2o-package/R/communication.R (hunk 1 above)
Comment (sebhrusen) [reply h2o-r/h2o-package/R/communication.R:1127]: right, I forgot that user is always allowed to make something silly

**Review by ledell — APPROVED** [review]
Thank you!

**Pull request merged**